Sources/Workspace/ManagedDependency.swift

```diff
@@ -172,12 +172,18 @@ extension Workspace.ManagedDependency: CustomStringConvertible {
 
 extension Workspace {
     /// A collection of managed dependencies.
-    final public class ManagedDependencies {
+    public struct ManagedDependencies {
```

Comment:
Turning `ManagedDependencies` into a `struct`. The alternative would be to make it an `actor` but since it conforms to `Collection`, it wouldn't be that easy.

There are only two methods of `ManagedDependencies` that are mutable – `add` and `remove`. Instead of mutating `self`, we can return copy of `Self` and do the mutation in `WorkspaceState`.